integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy

```diff
@@ -60,27 +63,77 @@ class OktaOrgCleaner {
         client.listGroups().stream()
-                .filter { it.getProfile().getName().matches(".*-${uuidRegex}.*")}
-                .forEach {
-                    log.info("\t ${it.getProfile().getName()}")
-                    it.delete()
-                }
+            .filter { it.getProfile().getName().matches(".*-${uuidRegex}.*") }
+            .forEach {
+                log.info("\t ${it.getProfile().getName()}")
+                it.delete()
+            }
 
         log.info("Deleting Group Rules:")
         client.listGroupRules().stream()
-                .filter { it.getName().matches("rule\\+${uuidRegex}.*")}
-                .forEach {
-                    GroupRule rule = it
-                    log.info("\t ${rule.getName()}")
-                    Util.ignoring(ResourceException) {
-                        rule.deactivate()
-                    }
-                    rule.delete()
+            .filter { it.getName().matches("rule\\+${uuidRegex}.*") }
+            .forEach {
+                GroupRule rule = it
+                log.info("\t ${rule.getName()}")
+                Util.ignoring(ResourceException) {
+                    rule.deactivate()
                 }
+                rule.delete()
+            }
 
         log.info("Deleting Policies:")
         client.listPolicies(PolicyType.OKTA_SIGN_ON.toString()).stream()
-            .filter { it.getName().matches("policy\\+${uuidRegex}.*")}
+            .filter { it.getName().matches("policy\\+${uuidRegex}.*") }
+            .forEach {
+                it.delete()
+            }
+
+        log.info("Deleting LinkedObjectDefinitions:")
+        client.listLinkedObjectDefinitions().stream()
+            .forEach {
+                it.setName(it.getPrimary().getName())
+                it.delete()
+            }
+
+        log.info("Deleting InlineHooks:")
+        client.listInlineHooks().stream()
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting EventHooks:")
+        client.listEventHooks().stream()
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting UserTypes:")
+        client.listUserTypes().stream()
+            .filter { !it.getDefault() }
+            .forEach {
+                it.delete()
+            }
+
+        log.info("Deleting AuthorizationServers:")
+        client.listAuthorizationServers().stream()
+            .filter { !it.getName().equals("default") }
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting IdentityProviders:")
+        client.listIdentityProviders().stream()
+            .filter()
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting SmsTemplates:")
+        client.listSmsTemplates().stream()
```

Comment:
The filtering should be done similar to the above code blocks were it delete the test created objects only.
The assumption here is that, anything created by a human would NOT have the format of `policy\\+${uuidRegex}.*` and anything created by a test would.  There are many things you could have created in your org and you may not want to delete them all.

Also, ideally, the test SHOULD clean up everything it creates. But even in a base case scenario this util class is useful for:
* Fixing tests that are not doing proper cleanup
* Cleaning up after tests under development
* Any random test failure that could have left resources behind.